### PR TITLE
Keep aad legacy code

### PR DESCRIPTION
**Pull request by chenrujun**
- Recover aad legacy code.
 - Add SPEL like `@ConditionalOnExpression("#{'${azure.active.directory.uri:notExist}' != 'notExist'}")` to make legacy code and new code not work at the same time.
 - Change new property prefix from 'azure.activedirectory' to 'azure.active.directory'.

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/implementation/AzureActiveDirectoryConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+
+@Configuration
+@ConditionalOnClass(ClientRegistrationRepository.class)
+@EnableConfigurationProperties(AzureActiveDirectoryProperties.class)
+@AutoConfigureAfter(AADOAuth2AutoConfiguration.class)
+@ConditionalOnExpression("#{'${azure.active.directory.uri:notExist}' != 'notExist'}")
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/implementation/AzureActiveDirectoryConfiguration.java:29]: Refs: https://stackoverflow.com/questions/49013862/how-to-get-spel-spring-el-default-to-null-expression-working-in-the-spring-xml

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**